### PR TITLE
Sync interface

### PR DESCRIPTION
**Pull request by wtrocki**
Raincatcher sync interface with minimal implementation. 

This sync module will use default storage (sync allows you to store files using direct mongo connection). Users will be able to register their dataset, but this can be also done directly in the client by specifying all sync options.

Leaving this open for discussion.

### CODE REVIEW

File: packages/syncapi/src/customSyncStorage.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { HasId, Store } from '@raincatcher/store';
```
Comment (wtrocki) [inline packages/syncapi/src/customSyncStorage.ts]: This was planned to be included for cases when custom storage is required. 
I'm not sure if we need any cloud init support at the moment until we want to use custom data store.

File: packages/syncapi/src/customSyncStorage.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    }
+
+    public function dataGetHandler(datasetId: any, queryParams: any, metadata: any, cb: any) {
+
+    }
+
+    public function dataDeleteHandler(datasetId: any, queryParams: any, metadata: any, cb: any) {
+
+    }
```
Comment (paolobueno) [inline packages/syncapi/src/customSyncStorage.ts]: A lot of `any`s here for now, but I think that's provisional until we use this and map closer to what sync feeds these functions.

Comment (wtrocki) [reply packages/syncapi/src/customSyncStorage.ts]: Going to remove this as we will probably need to change our aproach.

**Comment by wtrocki** [comment]
Sync should be implemented on the client side. We can still integrate this work for the cases.
Raincatcher should relay on sync and sync documentation. 
IMHO by default sync datastore should just execute init. All datastore actions can be done from client.

**Comment by wtrocki** [comment]
Merging back to be available as part of the module to test security

**Pull request merged**